lib/manager/npm/extract/monorepo.ts

```diff
@@ -19,15 +20,15 @@ export function detectMonorepos(
 ): void {
   logger.debug('Detecting Lerna and Yarn Workspaces');
   for (const p of packageFiles) {
+    const { packageFile, managerData = {} } = p;
     const {
-      packageFile,
-      npmLock,
-      yarnLock,
-      lernaDir,
       lernaClient,
+      lernaDir,
       lernaPackages,
+      npmLock,
+      yarnLock,
       yarnWorkspacesPackages,
-    } = p;
+    } = managerData as NpmManagerData;
```

Comment:
Same as above, the `packageFiles` parameter can be `Partial<PackageFile<NpmManagerData>>[]`, which means you can omit this assertion.